build.sbt

```diff
@@ -63,6 +62,7 @@ lazy val core = project
   .settings(
     moduleName := "scalafix-core",
     buildInfoSettingsForCore,
+    scalacOptions ++= syntheticsOn.value,
```

Comment:
instead of adding it to every scala2 project, can't you keep it in `ThisBuild` and remove it for the test-input? that would reduce the noise